eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Delegs.hs

```diff
@@ -70,7 +82,33 @@ data ConwayDelegsPredFailure era
       !(Map (RewardAcnt (EraCrypto era)) Coin)
   | -- | CERT rule subtransition Failures
     CertFailure !(PredicateFailure (EraRule "CERT" era))
-  deriving (Generic)
+  deriving (Generic, Typeable)
+
+instance NoThunks (PredicateFailure (EraRule "CERT" era)) => NoThunks (ConwayDelegsPredFailure era)
+
+instance
+  ( Era era
+  , EncCBOR (PredicateFailure (EraRule "CERT" era))
+  ) =>
+  EncCBOR (ConwayDelegsPredFailure era)
```

Comment:
FYI. Because these types are parametrized by `era`, it is possible to also easily add `FromCBOR` and `ToCBOR` as well. But, we can add them on the need to have basis